apple: Advanced settings and settings ux changes

roop opened the pull request:
This is a follow-up to #2477 -- I requested the apple changes to be not included in that PR.

This PR has:
  - A new tabbed UI for settings
  - Advanced settings for the dataplane configuration mentioned in #2477

roop committed:
Settings: Use a tabbed UI

In both macOS and iOS

roop committed:
Settings: Add support for advanced settings

roop committed:
Settings: UI for advanced settings

roop committed:
Settings: Save advanced settings to the tunnel

roop committed:
Tunnel: Apply logFilter from the configuration

roop committed:
Settings: Clean up the UI

Remove the "Logs" tab

roop committed:
Settings: macOS: Disable 'Apply' and 'Reset to Defaults' when applicable

roop committed:
Settings: Move Apply button next to the account field

roop committed:
Settings: Clean up badging and resetting to default

roop committed:
TunnelStore: Fix saving of advanced settings

roop committed:
Settings: Save defaults to disk when resetting advanced settings

roop committed:
Settings: macOS: Move footers above 'Apply' buttons

roop committed:
Settings: Fix 'Save' button disabledness

change submit label to "done"

roop committed:
Settings: updateTunnelAuthStatus when authBaseURL changes

jamilbk commented on swift/apple/FirezoneNetworkExtension/Adapter.swift:
```diff
@@ -90,24 +90,27 @@ public class Adapter {
-    private let logFilterString =
+    private let defaultLogFilter =
       "connlib_client_apple=info,firezone_tunnel=info,connlib_shared=info,connlib_client_shared=info,warn"
   #endif
```
These are duplicated from AdvancedSettings -- can we keep them defined in one place to make them easier to find and maintain?

roop replied:
This redundancy is a safeguard -- what if for some reason the main app hasn't set the log filter. Also, this can happen when the app is using a tunnel that was created by the previous version of the app (which doesn't set the logFilter field in the tunnel configuration) (this will not be an issue in production though).

roop replied:
On second thoughts, I do think I can handle this differently and avoid the duplication.

jamilbk replied:
> what if for some reason the main app hasn't set the log filter

This reminds me -- we should validate the logFilter string is non-blank / empty before saving, kinda like how we validate the other inputs.

roop replied:
> we should validate the logFilter string is non-blank / empty

We do validate it. The 'Apply' / 'Save' is disabled if the logFilter is empty.

roop replied:
The logFilter string duplication is fixed in commit [c250e02](https://github.com/firezone/firezone/pull/2540/commits/c250e025ef889d78fce9576000262947dbd66341).

jamilbk reviewed (changes requested):
LGTM, just a rec to have the log filter defined once so we can more easily test and change it.

jamilbk commented on swift/apple/FirezoneKit/Sources/FirezoneKit/Features/SettingsView.swift:
```diff
@@ -67,133 +105,341 @@ public struct SettingsView: View {
+    static let forAccount = "Your account ID is provided by your admin"
+    static let forAdvanced = """
+      Use the default values unless you know what you're doing. \
+      Changing them might disrupt access to your Firezone resources.
```
Perhaps we can make this text more prominent and also clarify that changing these fields is **not** supported? We're bound to have cowboys customize these, break and then open GitHub issues as a result 🤠 🙃 😃 

```suggestion
     WARNING: These settings are intended for internal debug purposes **only**. Changing these is not supported and will disrupt access to your Firezone resources.
```

roop replied:
Fixed in commit [d4e9555](https://github.com/firezone/firezone/pull/2540/commits/d4e955569d360d7dd5eb50d3e67513b222a88e62).

roop committed:
Settings: Change footnote warning text

Per review comments

roop committed:
Avoid duplicating the default connlib log filter string in code

Per review comments

To handle old tunnel configurations (created using code before this commit), we
set the log filter on the tunnel (if not already set) at the time we start the
tunnel.
